### PR TITLE
iziGhhu2: Hash eIDAS PIDs

### PR DESCRIPTION
**Pull request by jakubmiarka**

### CODE REVIEW

File: src/test/java/unit/uk/gov/ida/verifyserviceprovider/services/EidasAssertionServiceTest.java [hunk 1]
```diff
@@ -89,7 +92,8 @@ public void setUp() {
             levelOfAssuranceValidator,
             Optional.of(metadataResolverRepository),
             signatureValidatorFactory,
-            Optional.of(HUB_CONNECTOR_ENTITY_ID));
+            Optional.of(HUB_CONNECTOR_ENTITY_ID),
+            userIdHashFactory);
```
Comment (chrisholmes) [inline src/test/java/unit/uk/gov/ida/verifyserviceprovider/services/EidasAssertionServiceTest.java]: Could we get a test for this? Also, something in an integration test for it?

File: src/main/java/uk/gov/ida/verifyserviceprovider/services/EidasAssertionService.java [hunk 2]
```diff
@@ -32,6 +33,8 @@
     private final Optional<EidasMetadataResolverRepository> metadataResolverRepository;
     private final SignatureValidatorFactory signatureValidatorFactory;
     private final Optional<String> entityId;
+    private UserIdHashFactory userIdHashFactory;
+    private final AuthnContextFactory authnContextFactory = new AuthnContextFactory();
```
Comment (chrisholmes) [inline src/main/java/uk/gov/ida/verifyserviceprovider/services/EidasAssertionService.java]: We should probably DI this, but we can clean it up later.

File: src/main/java/uk/gov/ida/verifyserviceprovider/services/EidasAssertionService.java [hunk 3]
```diff
@@ -95,6 +105,12 @@ public LevelOfAssurance extractLevelOfAssuranceFrom(Assertion countryAssertion)
         }
     }
 
+    private Optional<AuthnContext> getAuthnContext(String uri) {
+        AuthnContext authnContext = authnContextFactory.mapFromEidasToLoA(uri);
+
+        return Optional.of(authnContext);
```
Comment (chrisholmes) [inline src/main/java/uk/gov/ida/verifyserviceprovider/services/EidasAssertionService.java]: This will through an Exception if the LOA is null

Comment (jakubmiarka) [reply src/main/java/uk/gov/ida/verifyserviceprovider/services/EidasAssertionService.java]: Yep, it should not be null at this point...  moreover the hashing actually requires LOA (AuthnContext) and will throw an exception too anyway, so I'm not 100% sure why we're passing this around as optional 🤷‍♂️

File: src/test/java/unit/uk/gov/ida/verifyserviceprovider/services/EidasAssertionServiceTest.java [hunk 4]
```diff
@@ -120,7 +113,7 @@ public void shouldCallValidatorsCorrectly() {
         List<Assertion> assertions = asList(
             anAssertionWithAuthnStatement(EIDAS_LOA_HIGH, "requestId").buildUnencrypted());
 
-        eidasAssertionService.translateSuccessResponse(assertions, "requestId", LevelOfAssurance.LEVEL_2, null);
+        eidasAssertionService.translateSuccessResponse(assertions, "requestId", LEVEL_2, null);
```
Comment (alan-gds) [inline src/test/java/unit/uk/gov/ida/verifyserviceprovider/services/EidasAssertionServiceTest.java]: I disagree with Codacy that this is an improvement, but it's probably not a battle worth fighting.

**Pull request merged**